### PR TITLE
[WIP] add a utility to convert pai share format to frameworkcontroller

### PR DESCRIPTION
**Pull request by xudifsd**
References:
* [Protocol description](https://github.com/Microsoft/pai/blob/50a427b258f1f37b171e63672f26ab1f0aeb6c3f/docs/marketplace-and-submit-job-v2/marketplace-and-submit-job-v2.md)
* [Current implementation for pai](https://github.com/Microsoft/pai/blob/master/src/rest-server/src/middlewares/jobV2.js)

We are building frameworkcontroller. The controller needs different job format to pai and pai-share, to utilize pai-share format, we need a converter to convert job description to one that frameworkcontroller can recognize.

Some fields in pai-share is not elegant enough, for example, current implementation for parsing uri from prerequisite is ugly and not general(Using , to partition multiple uri. Using github to trigger git clone).

Also I think we should merge this converter with [current converter](https://github.com/Microsoft/pai/blob/master/src/rest-server/src/middlewares/jobV2.js), this can reduce code we maintain and possible converter behavior conflicts.

Finished functionality of converter, needs to test some pai-share jobs in frameworkcontroller. Also need to add more test cases.

### CODE REVIEW

**Comment by yqwang-ms** [comment]
So, what is the roadmap for the conversion tool?
Seems this PR have not converted PAI Share API to CRD?

File: src/utilities/to_crd.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+    def to_crd(self):
+        desc = self.desc
+        if desc is None:
+            desc = "job of " + self.name
+
+        retry_policy = {"fancyRetryPolicy": False}
+        restart_policy = "Never"
+
+        result = {
+                "apiVersion": "launcher.microsoft.com/v1",
```
Comment (yqwang-ms) [inline src/utilities/to_crd.py]: API is changed:
apiVersion: frameworkcontroller.microsoft.com/v1
, please refer https://github.com/Microsoft/frameworkcontroller/tree/master/example/framework

I will update int bed controller now

**Comment by xudifsd** [comment]
> So, what is the roadmap for the conversion tool?
> Seems this PR have not totally converted PAI Share API to CRD.
> Such as cannot find volume related spec.

I'm not sure when to mount a volume, controller uses volume in NFS, but pai-share doesn't support NFS, I thinks this is what launcher's format doesn't compatible with pai-share's, since from pai-share's perspective, NFS is not reproducible.

For roadmap, I think we should be able to run all jobs described in pai-share, and later we should decide whether let this conversion tool and jobV2.js coexist or not. Maybe we will modify pai-share's format to make it more elegant, and being able to express NFS mount.

**Comment by yqwang-ms** [comment]
NFS is not reproducible? Do they use storage, like HDFS, HDFS is reproducible?
We can sync all these in the meeting.

**Comment by xudifsd** [comment]
I do not know their opinion, if we want to express NFS volume, we need to modify protocol of pai-share, like adding nfs prerequisite.

**Pull request closed without merging**